source/js/components/join/language-select.jsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+
+export default class LanguageSelect extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: "" };
+
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
please don't do this. The constructor should not be responsible for reshaping the instance bindings.